apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -129,8 +129,17 @@ pub mod oracle_process {
         loop {
             // update the latest block height
             latest_block = loop {
-                if let Ok(height) = oracle.eth_block_number().await {
-                    break height;
+                match oracle.eth_block_number().await {
+                    Ok(height) => break height,
+                    Err(error) => {
+                        tracing::warn!(
+                            ?error,
+                            "Couldn't get the latest Ethereum block number, \
+                             will keep trying"
+                        );
+                        tokio::time::sleep(std::time::Duration::from_secs(1))
+                            .await;
```

Comment:
This sleep here can sometimes cause a panic during shutdown, if `oracle.eth_block_number()` is still not being able to be gotten

```
2022-08-04T11:53:39.477439Z  INFO namada_apps::node::ledger: Broadcaster is no longer running.
2022-08-04T11:53:39.477650Z  INFO namada_apps::node::ledger: Anoma ledger node has shut down.
The application panicked (crashed).
Message:  A Tokio 1.x context was found, but it is being shutdown.
Location: /Users/main/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.19.2/src/time/driver/entry.rs:551

  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                                ⋮ 12 frames hidden ⋮
  13: core::panicking::panic_display::h61a0e95827ceaeba
      at /rustc/e092d0b6b43f2de967af0887873151bb1c0b18d3/library/core/src/panicking.rs:72
  14: tokio::time::driver::entry::TimerEntry::poll_elapsed::h2b0271330dcf0038
      at /Users/main/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.19.2/src/time/driver/entry.rs:551
       549 │     ) -> Poll<Result<(), super::Error>> {
       550 │         if self.driver.is_shutdown() {
       551 >             panic!("{}", crate::util::error::RUNTIME_SHUTTING_DOWN_ERROR);
       552 │         }
       553 │
  15: tokio::time::driver::sleep::Sleep::poll_elapsed::h0b5056a36f6b0bf3
      at /Users/main/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.19.2/src/time/driver/sleep.rs:403
       401 │         let coop = ready!(crate::coop::poll_proceed(cx));
       402 │
       403 >         let result = me.entry.poll_elapsed(cx).map(move |r| {
       404 │             coop.made_progress();
       405 │             r
  16: <tokio::time::driver::sleep::Sleep as core::future::future::Future>::poll::h8c2db896b3dbbcf3
      at /Users/main/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.19.2/src/time/driver/sleep.rs:435
       433 │         #[cfg(all(tokio_unstable, feature = "tracing"))]
       434 │         let _ao_poll_span = self.inner.ctx.async_op_poll_span.clone().entered();
       435 >         match ready!(self.as_mut().poll_elapsed(cx)) {
       436 │             Ok(()) => Poll::Ready(()),
       437 │             Err(e) => panic!("timer error: {}", e),
  17: namada_apps::node::ledger::ethereum_node::oracle::oracle_process::run_oracle_aux::{{closure}}::h49eefe10c43e0bac
      at /opt/workspace/heliax/repos/anoma/apps/src/lib/node/ledger/ethereum_node/oracle.rs:140
       138 │                              will keep trying"
       139 │                         );
       140 >                         tokio::time::sleep(std::time::Duration::from_secs(1))
       141 │                             .await;
       142 │                     }
```